Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchyBasedLinkResolver.swift

```diff
@@ -58,6 +58,26 @@ final class PathHierarchyBasedLinkResolver {
         }
     }
     
+    /// Traverse all symbols of the same kind that have collisions.
+    func traverseOverloadedSymbols(_ observe: (_ overloadedSymbols: [ResolvedTopicReference]) -> Void) {
+        let overloadableKinds = ["method", "type.method", "func", "init", "macro", "subscript", "func.op"]
+        
+        for (_, node) in pathHierarchy.lookup {
+            guard node.symbol != nil else { continue }
+            
+            // TODO: Once we know what data and what structure we need we should probably create a PathHierarchy API that hides those implementation details.
+            for (_ /*name*/, disambiguation) in node.children {
```

Comment:
I wonder if these nested loops should be a method on `PathHierarchy.Node` that calls a closure. This code knows a lot about the internal structure of these nodes and maybe should be encapsulated in that struct so we don't leak that structure.